src/libmongoc/src/mongoc/mongoc-cursor.c

```diff
@@ -1709,7 +1709,7 @@ _mongoc_cursor_prepare_getmore_command (mongoc_cursor_t *cursor,
       bson_append_int64 (command,
                          MONGOC_CURSOR_BATCH_SIZE,
                          MONGOC_CURSOR_BATCH_SIZE_LEN,
-                         batch_size);
+                         abs (_mongoc_n_return (cursor)));
```

Comment:
Just so I understand, this change is prompted by the following from [the corresponding spec change](https://github.com/mongodb/specifications/commit/9da20b4b28f99037215ddfd78f5b0df04897a0be), correct?

> When setting the **batchSize** on the **find** and **getMore** commands the value of **batchSize** **MUST** be based on the cursor limit calculations specified in the `CRUD`_ specification.

Is there a particular reason `abs()` is needed here? Looking in existing uses of `_mongoc_n_return`, it looks like it was only used in `mongoc-cursor-legacy.c` for legacy OP_QUERY and OP_GET_MORE operations. In that case, I suppose a negative value was relevant (at least for [OP_QUERY](https://docs.mongodb.com/manual/reference/mongodb-wire-protocol/#op_query)) in that it'd request as many documents and immediately close the cursor.

Reply:
I'm not sure about the exact motivations for using `abs()`, this merely restores functionality to what it was before I worked around the 5.0 issue. We'll be revisiting this logic in DRIVERS-1448, which will be done once we drop support for MongoDB < 3.6, which allows us to remove all `OP_QUERY` and `OP_GET_MORE` logic, including having the driver track the cursor limit.